### ISSUE TITLE
Add `TinyInt / i8` type

### ISSUE THREAD

**Issue by iajoiner** [spaceandtimelabs/sxt-proof-of-sql#154]
# Background and Motivation

We already have `SmallInt / i16`, `Int / i32`, and `BigInt / i64` data types supported. Adding an additional `TinyInt / i8` data type would be helpful to some users and would additionally have (minor) performance benefits with those columns.

# Changes Requested

- [ ] Add a `TinyInt` variant to the various `*Column*` enum types.
- [ ] Add a branch for handling `TinyInt` everywhere that `SmallInt`, `Int`, and `BigInt` are handled.
- [ ] Add unit tests covering the additional code.

NOTE: be sure keep individual PRs small and self-contained. All net-new code should be tested within the PR.

## EDIT: clarification

I've seen several attempts at this issue, and most have the following concern: there aren't any tests around using +, -, =, >, *, etc. on TinyInt columns.

In other words, I'm not convinced that the following query will execute properly if `a`, `b`, and `c` are TinyInt columns:
```sql
SELECT a*b+b+c FROM table WHERE a>b OR c=4
```

It might be the case that there is code in place to do this. However, there should be unit tests providing evidence that this is the case.

At a minimum, in order to close this ticket we need to have at least one integration test showing end-to-end behavior with `TinyInt` columns and a non-trivial query that requires computation involving `TinyInt`.

**Comment by drcapybara** [comment]
Thanks for adding this, should be straightforward enough to add it to the column types and then fill everything out. A good strategy is to do a grep/search in the repo for something like "SmallInt" to find all of the places where it will need to be added.

**Comment by JayWhite2357** [comment]
\bounty $500

**Comment by Dustin-Ray** [comment]
@c0d33ngr please feel free to reach out with any questions, there are some nuances to this change that might warrant clarification.

**Comment by c0d33ngr** [comment]
Yes, I'll if I need clarification

**Comment by Ashu999** [comment]
/attempt https://github.com/spaceandtimelabs/sxt-proof-of-sql/issues/154

**Comment by JayWhite2357** [comment]
I added a clarification to the description that I many of these attempts need to handle.
